src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -20,17 +21,50 @@ export default class SimidMapCreative extends BaseSimidCreative {
   }
 
   /** @override */
-  onStart(eventData){
-    super.onStart(eventData);
-    //ToDo(juliareichel@): handle invalid JSON and param errors
-    const adParams = JSON.parse(this.creativeData.adParameters);
+  onInit(eventData) {
+    this.validateAndParseAdParams_(eventData);
+    super.updateInternalOnInit(eventData);
+  }
+
+  /**
+   * Checks validity of ad parameters and rejects with proper message if invalid.
+   * @param eventData an object that contains information details for a particular event
+   *   such as event type, unique Ids, creativeData and environmentData.
+   * @private 
+  */ 
+  validateAndParseAdParams_(eventData) {
+    this.creativeData = eventData.args.creativeData;
```

Comment:
If you swap the validate and updateInternal calls in onInit, you can remove this line

Reply:
swapped and removed